### DISCUSSION TITLE
how to send args in post_to_main_thread.

### DISCUSSION THREAD

**Discussion by MeyerBuaharon** [isl-org/Open3D#3938] [Q&A] [answered]
Hey,
I'm using open3d and I've created a thread that runs in the background and give me the next point cloud:
`gui.Application.instance.post_to_main_thread(window, next_point_cloud)`

Is there a way to send args to my `next_point_cloud` function while using post_to_main_thread?

**Comment by hduonggithub** [comment, marked as answer]
Line 110-111
https://github.com/isl-org/Open3D/blob/master/examples/python/gui/add-geometry.py

**Reply by MeyerBuaharon** [reply]
Thanks for the response,
If I wanted to send another parameter to add_sphere, so how do I send the parameter in the `post_to_main_thread`?

```
def add_sphere(self, another_parameter):
        self._id += 1
        mat = rendering.Material()
        ....

```

```
gui.Application.instance.post_to_main_thread(
                    self.window, self.add_sphere)  <- how I add here another parameter?
```

**Reply by hduonggithub** [reply]
You can make global variables for other inputs as you cannot pass any other parameters. As you see the self._id is a global variable that can be seen in add_sphere function.

**Reply by MeyerBuaharon** [reply]
Got it, Thanks
